lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Context.java

```diff
@@ -30,15 +30,27 @@
 import static io.trino.filesystem.s3.S3FileSystemConstants.EXTRA_CREDENTIALS_SESSION_TOKEN_PROPERTY;
 import static java.util.Objects.requireNonNull;
 
-record S3Context(int partSize, boolean requesterPays, S3SseType sseType, String sseKmsKeyId, Optional<AwsCredentialsProvider> credentialsProviderOverride, ObjectCannedAcl cannedAcl)
+record S3Context(
+        int partSize,
+        boolean requesterPays,
+        S3SseType sseType,
+        String sseKmsKeyId,
+        S3SseCustomerKey sseCustomerKey,
```

Comment:
So can we have something like SseContext which could capture these information - Extraction could be a prep commit. WDYT ?

Reply:
extracted